### PR TITLE
Refactoring tidy...

### PR DESCRIPTION
**Pull request by ghost**
Improved tidy's code for efficiency (which now shows progress while checking through files)

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9649)

### CODE REVIEW

**Comment by Wafflespeanut** [comment]
(sigh I've forgotten about the "faster" flag)

File: python/tidy.py [hunk 1]
```diff
@@ -48,17 +48,31 @@
...
-    for pattern in ignored_files:
-        if fnmatch.fnmatch(file_name, pattern):
-            return False
-    return True
+def filter_files(start_dir, faster):
+    args = 'find %s -type f | wc -l' % start_dir
+    tree_count = subprocess.Popen(args, shell=True, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+    total_files, progress = '???', '???'
+    for idx, file_name in enumerate(get_file_list(start_dir, faster)):
+        if type(total_files) is int:
```
Comment (frewsxcv) [inline python/tidy.py]: I'd recommend `isinstance(total_files, (int, long))` here

Comment (Wafflespeanut) [inline python/tidy.py]: Nice! :)

**Comment by Wafflespeanut** [comment]
actually, I should add another flag for not showing the progress (at least for the sake of Travis) - ideas?

**Comment by Wafflespeanut** [comment]
Added a flag for not showing the progress (for the sake of Travis)

**Comment by Wafflespeanut** [comment]
This still needs some work (firstly, let's print the progress percent alone)

**Comment by Wafflespeanut** [comment]
@jdm I think I've reduced the time taken by tidy. If everything goes well, then #8969 will be fixed :)

Previously, tidy took ~8 mins in my machine - now, it takes only ~5 mins, in which tidy's checks only take ~20 seconds, whereas the WPT lint takes ~4.5 mins or something

**Comment by Wafflespeanut** [comment]
@jdm I think I've reduced the time taken by tidy. If everything goes well, then #8969 will be fixed :)

Previously, tidy took ~8 mins in my machine - now, it takes only ~5 mins, in which tidy's checks only take ~20 seconds, whereas the WPT lint takes ~4.5 mins or something

File: python/tidy.py [hunk 2]
```diff
@@ -536,30 +554,35 @@ def get_file_list(directory, only_changed_files=False):
         # also check untracked files
         args = ["git", "ls-files", "--others", "--exclude-standard", directory]
         file_list += subprocess.check_output(args)
-        return (os.path.join(".", f) for f in file_list.splitlines())
+        for f in file_list.splitlines():
+            yield os.path.join('.', f)
+    elif exclude_dirs:
+        for root, dirs, files in os.walk(directory, topdown=True):
+            # modify 'dirs' in-place so that we don't do unwanted traversals in excluded directories
+            dirs[:] = [d for d in dirs if not any([os.path.join(root, d).startswith(name) for name in ignored_dirs])]
```
Comment (frewsxcv) [inline python/tidy.py]: `[d for d in dirs if not any(os.path.join(root, d).startswith(name) for name in ignored_dirs)]`

File: python/tidy.py [hunk 3]
```diff
@@ -21,44 +20,57 @@
...
+    if progress:
+        file_iter = progress_wrapper(file_iter)
+    for file_name in file_iter:
+        if os.path.basename(file_name) == "Cargo.lock":
+            yield file_name
+        if ".#" in file_name:
+            continue
+        if os.path.splitext(file_name)[1] not in filetypes_to_check:
+            continue
+        if any([file_name.startswith(ignored_file) for ignored_file in ignored_files]):
```
Comment (frewsxcv) [inline python/tidy.py]: `if any(file_name.startswith(ignored_file) for ignored_file in ignored_files):

**Comment by Wafflespeanut** [comment]
Oh, yeah, generators, forgot that :)

File: python/servo/testing_commands.py [hunk 4]
```diff
@@ -94,10 +94,12 @@ def run_test(self, prefix, args=[], release=False):
     @CommandArgument('--release', default=False, action="store_true",
                      help="Run with a release build of servo")
     @CommandArgument('--faster', default=False, action="store_true",
-                     help="Only check changed files and skip the WPT lint")
-    def test(self, params, render_mode=DEFAULT_RENDER_MODE, release=False, faster=False):
+                     help="Only check changed files and skip the WPT lint in tidy")
+    @CommandArgument('--no-progress', default=False, action="store_true",
+                     help="Don't show progress for tidy")
```
Comment (jdm) [inline python/servo/testing_commands.py]: I don't think we need this (or faster). `./mach test` is a special case, and we can choose appropriate defaults for it.

**Comment by jdm** [comment]
-S-awaiting-review +S-needs-code-changes 
This generally looks fine, but I left a couple comments.

---

Reviewed 1 of 3 files at r2, 1 of 1 files at r3.
Review status: 2 of 3 files reviewed at latest revision, 5 unresolved discussions.

---

*[python/tidy.py, line 45 \[r3\]](https://reviewable.io:443/reviews/servo/servo/9649#-KBK6BCJN2bkuBFZAXYP:-KBK6BCJN2bkuBFZAXYQ:372420158) ([raw file](https://github.com/servo/servo/blob/6e2568752414e4b9cfa4ac79b1b914f5693929e3/python/tidy.py#L45)):*
This isn't the same as what existed before. We no longer ignore files that start with a . in the root directory.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9649#-:-KBKCMkgUODD1lWCVfXA:-1189603826)*

**Comment by Wafflespeanut** [comment]
Review status: 2 of 3 files reviewed at latest revision, 5 unresolved discussions.

---

*[python/servo/testing_commands.py, line 99 \[r3\]](https://reviewable.io:443/reviews/servo/servo/9649#-KAzrA_nbIBgMG1WCoHh-r3-99:-KBMKzBRw8sCVmcjpNSI:-1723455086) ([raw file](https://github.com/servo/servo/blob/6e2568752414e4b9cfa4ac79b1b914f5693929e3/python/servo/testing_commands.py#L99)):*
Okay, shall we remove it while refactoring the command as a part of #9544 ?

---

*[python/tidy.py, line 45 \[r3\]](https://reviewable.io:443/reviews/servo/servo/9649#-KBK6BCJN2bkuBFZAXYP:-KBMJPR3TKO5qRuTKTOF:-266320828) ([raw file](https://github.com/servo/servo/blob/6e2568752414e4b9cfa4ac79b1b914f5693929e3/python/tidy.py#L45)):*
Hmm, yeah, I missed that. It didn't show anything in my test run (probably because we only have `.travis.yml` and `.gitignore`, which is prevented by `filetypes_to_check`) ...

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9649)*

**Comment by jdm** [comment]
@bors-servo: r+

---

Reviewed 1 of 3 files at r2, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[python/servo/testing_commands.py, line 99 \[r3\]](https://reviewable.io:443/reviews/servo/servo/9649#-KAzrA_nbIBgMG1WCoHh-r3-99:-KBNsbe4x0z-0VC0gikn:-485110604) ([raw file](https://github.com/servo/servo/blob/6e2568752414e4b9cfa4ac79b1b914f5693929e3/python/servo/testing_commands.py#L99)):*
Sure.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9649#-:-KBNsegUWDUCWXq0zv94:-317921516)*

**Comment by Wafflespeanut** [comment]
@bors-servo retry

#8120

**Pull request merged**